module/VuFindAdmin/src/VuFindAdmin/Controller/TagsController.php

```diff
@@ -112,10 +113,17 @@ public function listAction()
     {
         $view = $this->createViewModel();
         $view->setTemplate('admin/tags/list');
-        $view->uniqueTags      = $this->getUniqueTags()->toArray();
-        $view->uniqueUsers     = $this->getUniqueUsers()->toArray();
-        $view->uniqueResources = $this->getUniqueResources()->toArray();
-        $view->results = $this->getResourceTags(false);
+        $view->uniqueTags = $this->getUniqueTags();
+        $view->uniqueUsers = $this->getUniqueUsers();
+        $view->uniqueResources = $this->getUniqueResources();
+        $page = intval($this->getParam('page', false, '1'));
+        $view->results = $this->getDbService(ResourceTagsServiceInterface::class)->getResourceTagsPaginator(
```

Comment:
This is another case of (the same) database services being retrieved by a controller (multiple times) as mentioned in [#3701](https://github.com/vufind-org/vufind/pull/3701#discussion_r1618452188).